inc/providers/class-wp-sitemaps-posts.php

```diff
@@ -136,24 +150,72 @@ public function get_url_list( $page_num, $post_type = '' ) {
 	 * @param string $post_type Optional. Post type name. Default empty.
 	 * @return int Total number of pages.
 	 */
-	public function max_num_pages( $post_type = '' ) {
+	public function get_max_num_pages( $post_type = '' ) {
 		if ( empty( $post_type ) ) {
 			return 0;
 		}
 
-		$query = new WP_Query(
+		/**
+		 * Filters the max number of pages before it is generated.
+		 *
+		 * Passing a non-null value will effectively short-circuit the generation,
+		 * returning that value instead.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param int $max_num_pages The maximum number of pages. Default null.
+		 * @param string $post_type Post type name.
+		 */
+		$max_num_pages = apply_filters( 'wp_sitemaps_posts_pre_max_num_pages', null, $post_type );
+
+		if ( null !== $max_num_pages ) {
+			return $max_num_pages;
+		}
+
+		$args = $this->get_posts_query_args( $post_type );
+		$args['paged'] = 1;
+		$args['fields'] = 'ids';
+		$args['no_found_rows'] = false;
+
+		$query = new WP_Query( $args );
+
+		return isset( $query->max_num_pages ) ? $query->max_num_pages : 1;
+	}
+
+	/**
+	 * Returns the query args for retrieving posts to list in the sitemap.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $post_type Post type name.
+	 * @return array $args Array of WP_Query arguments.
+	 */
+	protected function get_posts_query_args( $post_type ) {
+		/**
+		 * Filters the query arguments for post type sitemap queries.
+		 *
+		 * @see WP_Query for a full list of arguments.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param array  $args      Array of WP_Query arguments.
+		 * @param string $post_type Post type name.
+		 */
+		$args = apply_filters(
+			'wp_sitemaps_posts_query_args',
 			array(
-				'fields'                 => 'ids',
 				'orderby'                => 'ID',
 				'order'                  => 'ASC',
 				'post_type'              => $post_type,
 				'posts_per_page'         => wp_sitemaps_get_max_urls( $this->object_type ),
```

Comment:
is there any reason, why not directly call `$this->get_max_num_pages`?

Reply:
Those are two different things.

`get_max_num_pages` returns the maximum amount of pages.
`wp_sitemaps_get_max_urls` returns the maximum amount of posts on a single page

Not to mention that `get_max_num_pages` calls this function here. You don't wanna cause an infinite loop :-)

Reply:
Oops, totally misread that! Sorry!

Reply:
No worries! 👍